src/arduino/arduino.ts

```diff
@@ -140,8 +140,8 @@ export class ArduinoApp {
             }
         }
 
-        const appPath = path.join(ArduinoWorkspace.rootPath, dc.sketch);
-        const args = ["--upload", "--board", boardDescriptor];
+        const appPath = this.isArduinoCli() ? ArduinoWorkspace.rootPath : path.join(ArduinoWorkspace.rootPath, dc.sketch);
+        const args = this.isArduinoCli() ? ["upload", "-b", boardDescriptor] : ["--upload", "--board", boardDescriptor];

```

Comment:
To maintain matching behavior with IDE version, this should probably use the compile comand with --upload flag, ensuring that code is both compiled and uploaded

Reply:
I added a new compile arg to upload(), set to true by default. This way we keep the functionality the same as the CLI, but also provide an option to upload without compiling